### ISSUE TITLE
Errorneous suggestion inside type class method definition

### ISSUE THREAD

**Issue by theindigamer** [ndmitchell/hlint#567]
In the following code

    import Data.Ord (comparing)

    data P a = P {x, y :: Int, blah :: a}

    instance Eq (P a) where
      p1 == p2 = compare p1 p2 == EQ

    instance Ord (P a) where
      compare p1 p2 = comparing x p1 p2 <> comparing y p1 p2

there is a suggestion to replace `compare p1 p2 == EQ` with `p1 == p2`. I don't know if `hlint` keeps track of sufficient context to be able to realize that this suggestion is incorrect.

**Comment by ndmitchell** [comment]
Yep, it currently lacks the context to do that... It is an issue that has come up before, so worth thinking about.
